### PR TITLE
realtek-rtl838x: add support for D-Link DGS-1210-10P

### PR DESCRIPTION
**Pull request by AiyionPrime**
> as well as the target itself

- [x] Must be flashable from vendor firmware
  - [x] Web interface (instructions are somewhat complex, not recommended)
  - [x] TFTP
  - ~~Other: <specify>~~
- [x] Must support upgrade mechanism
  - [x] Must have working sysupgrade
    - [x] Must keep/forget configuration (`sysupgrade [-n]`, `firstboot`)
  - [x] Gluon profile name matches autoupdater image name
        (`lua -e 'print(require("platform_info").get_image_name())'`)
- [x] Reset/WPS/... button must return device into config mode
- [x] Primary MAC address should match address on device label (or packaging)
      (https://gluon.readthedocs.io/en/latest/dev/hardware.html#hardware-support-in-packages)
  - When re-adding a device that was supported by an earlier version of Gluon, a
    factory reset must be performed before checking the primary MAC address, as
    the setting from the old version is not reset otherwise.
- Wired network
  - [x] should support all network ports on the device (ports nine and ten *sfp* reportedly work in OpenWrt, but not tested)
  - [x] must have correct port assignment (WAN/LAN)
    - On devices supplied via PoE, there is usually no explicit WAN/LAN labeling on the hardware.
      The PoE input should be the WAN port in this case.
- Wireless network (if applicable)
  - ~~Association with AP must be possible on all radios~~
  - ~~Association with 802.11s mesh must work on all radios~~ 
  - ~~AP+mesh mode must work in parallel on all radios~~
- LED mapping
  - Power/system LED
    - [x] Lit while the device is on
    - [x] Should display config mode blink sequence 
          (https://gluon.readthedocs.io/en/latest/features/configmode.html)
  - Radio LEDs
    - ~~Should map to their respective radio~~
    - ~~Should show activity~~
  - Switch port LEDs
    - [x] Should map to their respective port (or switch, if only one led present) 
    - [x] Should show link state and activity
- Outdoor devices only:
  - ~~Added board name to `is_outdoor_device` function in `package/gluon-core/luasrc/usr/lib/lua/gluon/platform.lua`~~
- Docs:
  - [x] Added Device to `docs/user/supported_devices.rst`

### CODE REVIEW

**Comment by blocktrron** [comment]
The target lacks USB support, so this package is not available, thus can not be dis- nor enabled.

Rename the line configuring this package in `targets/generic` to `try_config`

**Comment by AiyionPrime** [comment]
That looks promising, thanks @blocktrron.
I'm currently a few hundred km from the switch; I think I'll get the next round of testing done till noon tomorrow.

**Comment by AiyionPrime** [comment]
Looks like that device does not support isolation either?
Tested this before @herbetom and I met in mumble on monday.
Can do the checklist tomorrow afternoon, but I'm starting to feel the need for an information-source/a script that can tell, whether this device is expected to do port isolation, or not...

#2679

**Comment by herbetom** [comment]
> but I'm starting to feel the need for an information-source/a script that can tell, whether this device is expected to do port isolation, or not...

It's actually relatively easy. 

no isolation if one of the following applies:
- unmaged switch ports
- switch, but it's an target still configured via swconfig

In the following cases isolation is expected to work between all ports of that type and ports of other types:

- DSA Switch Ports (happens on a per target base)
- dedicated ports (like the ethernet on an APU)
- VLANs

**Comment by AiyionPrime** [comment]
Said switch is managed, realtek is DSA since 21.02.
At least in hannover this switch does not isolate bridgeports, but does when used with vlans instead.

**Comment by AiyionPrime** [comment]
Everything except the reset button is working fine, was dropped as invalid in

https://github.com/openwrt/openwrt/commit/61a3d0075b151f6742862e123fdff1897d876176#diff-3fa215c7c9438012008b63350434341f8a2bc01b514837ecc11263e928822dd2L42-R49

I'll review his GPIO suspicion later...

**Comment by AiyionPrime** [comment]
Patch has been submitted.
https://patchwork.ozlabs.org/project/openwrt/patch/20221202135516.4065670-1-git@aiyionpri.me/

**Comment by AiyionPrime** [comment]
Local testing is done, Buttons and LEDs work as intended.
Will be merged, once the actions are done.

**Comment by heini66** [comment]
a 1210-24 rv d1 is a roommate. which image should i flash? ;)


Am Sa., 10. Dez. 2022 um 17:52 Uhr schrieb David Bauer <
***@***.***>:

**Comment by AiyionPrime** [comment]
Good evening @heini66. You should avoid flashing Images onto devices, that are not supported.

I'm out for tonight, but will look into the next steps for your device tomorrow.
Don't get you hopes up to early though; there are only few revisions supported by OpenWrt yet.
See you.

**Comment by AiyionPrime** [comment]
If you can please provide a vendor link or a complete version string for your device.
I'm not sure

> a 1210-24 rv d1

is really a thing.

**Comment by heini66** [comment]
https://ftp.dlink.de/dgs/dgs-1210/driver_software/dgs-1210_fw_revd_422b00X_all_multi_20220502.zip
https://eu.dlink.com/de/de/-/media/business_products/dgs/dgs-1210/datasheet/dgs_1210_series_g1_datasheet_en_eu.pdf

Am So., 18. Dez. 2022 um 00:10 Uhr schrieb Jan-Niklas Burfeind <
***@***.***>:

**Comment by heini66** [comment]
24 * rj-45 ports, 4 sfp, no poe, hw rev d1

Am So., 18. Dez. 2022 um 00:24 Uhr schrieb heini66 ***@***.***>:

**Comment by AiyionPrime** [comment]
I think this would be the device you describe, not sure whether the developer or dlink introduced the two different names, sorry.

https://git.openwrt.org/?p=openwrt/openwrt.git;a=commit;h=b5bd945733a7ed235777c2ceaa0ff27dc779630f

The revsions F1 and F2 (and possibly newer ones newer as well) are supported by OpenWrt.
Your D1 has very likely a different SoC and is not.

https://wikidevi.wi-cat.ru/D-Link_DGS-1210-24_rev_A1

File: targets/realtek-rtl838x [hunk 1]
```diff
@@ -0,0 +1,5 @@
+-- D-Link
+
+device('d-link_dgs-1210-10p', 'd-link_dgs-1210-10p', {
```
Comment (neocturne) [inline targets/realtek-rtl838x:3]: Is this device name correct? For most targets it should be impossible for an underscore to appear.

**Pull request merged**